dissect/target/helpers/ssh.py

```diff
@@ -0,0 +1,187 @@
+import base64
+import binascii
+
+from dissect import cstruct
+
+
+class SSHPrivateKey:
+    """
+    A class to parse (OpenSSH-supported) SSH private keys.
+
+    OpenSSH supports three types of keys:
+    * RFC4716 (default)
+    * PKCS8
+    * PEM
+    """
+
+    # RFC4716/OpenSSH Format
+    RFC4716_MARKER_START = b"-----BEGIN OPENSSH PRIVATE KEY-----"
+    RFC4716_MARKER_END = b"-----END OPENSSH PRIVATE KEY-----"
+    RFC4716_MAGIC = b"openssh-key-v1\x00"
+    RFC4716_PADDING = b"\x01\x02\x03\x04\x05\x06\x07"
+    RFC4716_NONE = b"none"
+    RFC4716_STRUCT = """
```

Comment:
If you move this into the global scope, can you also initialize the cstruct instance in the global scope? That ensures that the cstruct instance is only ever loaded once, which can benefit performance.